### PR TITLE
Validation ordering filter

### PR DESCRIPTION
**Pull request by ticosax**
Follow up for https://github.com/carltongibson/django-filter/pull/647#discussion_r103686886

Build on top of #648, (I will rebase, once merged. (if merged. (I hope.)))

### CODE REVIEW

File: tests/test_filterset.py [hunk 1]
```diff
@@ -687,33 +689,50 @@ def setUp(self):
 
     def test_fields_set(self):
         class F(FilterSet):
+            o = OrderingFilter(fields=('username', 'status'))
```
Comment (rpkilby) [inline tests/test_filterset.py]: I would create separate test methods for ordering - too much is under test here.

File: tests/test_filterset.py [hunk 2]
```diff
@@ -687,33 +689,50 @@ def setUp(self):
...
+
         f = F({'username': 'alex', 'status': 1}, queryset=self.qs)
         self.assertEqual(f.qs.count(), 1)
         self.assertQuerysetEqual(f.qs, [self.alex.pk], lambda o: o.pk)
 
+        f = F({'o': 'username'}, queryset=self.qs)
+        with self.assertRaises(ValidationError):
+            f.qs.count()
+
+        f = F({'o': 'username,status'}, queryset=self.qs)
```
Comment (rpkilby) [inline tests/test_filterset.py]: What about descending order?

File: django_filters/filterset.py [hunk 3]
```diff
@@ -50,13 +51,19 @@ def all_valid(fieldset):
             count = len([i for i in fieldset if cleaned_data.get(i)])
             return 0 < count < len(fieldset)
 
+        def is_ordering_invalid(fieldset):
+            if ordering_field is None or not form.cleaned_data.get(ordering_field):
+                return
+            count = len([i for i in fieldset if i in form.cleaned_data[ordering_field]])
+            return 0 < count < len(fieldset)
```
Comment (rpkilby) [inline django_filters/filterset.py]: The intent is to fail if any field in the fieldset is not present in the ordering field's data. Checking count bounds does not clearly convey this intent. Maybe something like:

```python
ordering_data = form.cleaned_data[ordering_field]

# tests for presence in ordering data, returns list of true/false
together_ordering = [field in ordering_data for field in fieldset]

if any(together_results):
    return all(together_results)
return True  # default to success
```

File: tests/test_filterset.py (hunk 2 above)
Comment (ticosax) [inline tests/test_filterset.py]: good point !

**Comment by carltongibson** [comment]
Again, I don't want to add this. I don't see the behaviour, the code here, or the syntax to declare it, as that clear. 

I'd much rather steer people towards implementing the base `Form` class. That seems like the preferable _one obvious way to do it_.

**Pull request closed without merging**